### PR TITLE
Encode class labels alphabetically (sklearn convention)

### PR DESCRIPTION
**Pull request by weihaokong**
Add an `"alphabetical"` mode to `CategoricalOrdinalEncoder` (sort categories ascending) and use it for the label encoder, so `TabFMClassifier.classes_` and `predict_proba` column order follow scikit-learn's `LabelEncoder` convention (classes sorted ascending) instead of order-of-appearance.

### CODE REVIEW

File: tabfm/src/classifier_and_regressor.py [hunk 1]
```diff
@@ -148,6 +150,10 @@ def fit(self, X: Any, y: Any = None) -> "CategoricalOrdinalEncoder":
         uniques = [u for u in uniques if str(u) != "nan"]
         # Filter rare categories
         uniques = [u for u in uniques if u not in rare_cats]
+        if self.mode == "alphabetical":
```
Comment (siriuz42) [inline tabfm/src/classifier_and_regressor.py]: nit: use two `elif` for "alphabetical" and "appearance", and add an `else` and a `raise ValueError`.

**Review by siriuz42 — APPROVED** [review]

File: tabfm/src/classifier_and_regressor.py (hunk 1 above)
Comment (weihaokong) [reply tabfm/src/classifier_and_regressor.py]: thanks!
